src/ExtOperator.actor.cpp

```diff
@@ -554,9 +554,7 @@ ACTOR static Future<Void> doPopActor(Reference<IReadWriteContext> cx,
 					scx->getSubContext(kp)->clearDescendants();
 					scx->clear(kp);
 					if (oldNumber > 0) {
-						if (next.getSortType() == DVTypeCode::NULL_ELEMENT) {
-							// no-op, so our sparse arrays stay sparse
-						} else if (next.isSimpleType()) {
```

Comment:
`NUL_ELEMENT` is a simple type. It would set the element in the array. Is that the behavior you are looking for?

Reply:
Yes exactly. It would set the element in the array, in other words it's NOT storing the array as a sparse array.

Reply:
I should have been more clear about what this PR is about in the title.

Reply:
Isn't it better to store the array as sparse? If we insert an array element at index 1000, it would insert 1000 keys to represent nulls in previous keys. Can you explain why we shouldn't store arrays as sparse?

Reply:
Storing the array as sparse arrays require changes in `doPathExpansion`. I did have a hack-y way to achieve that, but I think the actual change could potentially be harder than this.

Also I feel it's fair to assume the sparse array is not a common thing when using Mongo and if the user really wants the support of that, we can then put together something to achieve that.